### PR TITLE
[wolfssl] Add feature flags `dtls` and `secret-callback`

### PR DESCRIPTION
**Pull request by laugengebaeck**
This adds feature flags that enable DTLS and `HAVE_SECRET_CALLBACK` (to provide TLS keys for debugging) to the `wolfssl` port.

(Context: In [eulynx-live/librasta](https://github.com/eulynx-live/librasta), we use these two features of WolfSSL. We recently started migrating our dependency management to vcpkg and adding these features to the port would help us make the migration succeed).

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by jimwang118** [comment]
The following error occurred in the local test.
```
LINK Pass 1: command "C:\PROGRA~1\MICROS~4\2022\ENTERP~1\VC\Tools\MSVC\1436~1.325\bin\Hostx64\x64\link.exe CMakeFiles\wolfssl.dir\wolfcrypt\src\hmac.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\hash.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\cpuid.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\kdf.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\random.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\sha256.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\rsa.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\sp_int.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\aes.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\des3.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\sha.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\sha512.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\sha3.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\logging.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\wc_port.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\error.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\wc_encrypt.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\signature.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\wolfmath.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\memory.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\dh.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\asn.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\coding.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\poly1305.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\md5.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\pwdbased.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\pkcs12.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\chacha.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\chacha20_poly1305.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\integer.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\ecc.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\pkcs7.c.obj CMakeFiles\wolfssl.dir\src\internal.c.obj CMakeFiles\wolfssl.dir\src\wolfio.c.obj CMakeFiles\wolfssl.dir\src\keys.c.obj CMakeFiles\wolfssl.dir\src\ssl.c.obj CMakeFiles\wolfssl.dir\src\ocsp.c.obj CMakeFiles\wolfssl.dir\src\tls.c.obj CMakeFiles\wolfssl.dir\src\tls13.c.obj CMakeFiles\wolfssl.dir\src\crl.c.obj CMakeFiles\wolfssl.dir\wolfcrypt\src\cryptocb.c.obj /out:wolfssl.dll /implib:wolfssl.lib /pdb:wolfssl.pdb /dll /version:35.5 /machine:x64 /nologo /debug /INCREMENTAL ws2_32.lib crypt32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\wolfssl.dir/intermediate.manifest CMakeFiles\wolfssl.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library wolfssl.lib and object wolfssl.exp
internal.c.obj : error LNK2019: unresolved external symbol DoClientHelloStateless referenced in function DoClientHello
internal.c.obj : error LNK2019: unresolved external symbol DtlsResetState referenced in function DoClientHello
internal.c.obj : error LNK2019: unresolved external symbol DtlsIgnoreError referenced in function DoClientHello

internal.c.obj : error LNK2019: unresolved external symbol DtlsSetSeqNumForReply referenced in function DoClientHello

wolfssl.dll : fatal error LNK1120: 4 unresolved externals

ninja: build stopped: subcommand failed.

```

**Comment by laugengebaeck** [comment]
@jimwang118 How can I reproduce this? For me, the build process seems to work fine.

**Comment by jimwang118** [comment]
> @jimwang118 How can I reproduce this? For me, the build process seems to work fine.

I use the command: `.\vcpkg.exe install wolfssl[*]:x64-windows` to install.

**Comment by laugengebaeck** [comment]
@jimwang118 Thanks! I now identified and fixed the issue, so that Wolfssl should compile successfully on Windows. Could you review again, please?

**Comment by jimwang118** [comment]
> @jimwang118 Thanks! I now identified and fixed the issue, so that Wolfssl should compile successfully on Windows. Could you review again, please?

OK, please resolve the existing code conflicts and I will review the code as soon as possible.

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

File: ports/wolfssl/portfile.cmake [hunk 1]
```diff
@@ -9,6 +9,17 @@ vcpkg_from_github(
       fix-windows-build.patch # From https://github.com/wolfSSL/wolfssl/pull/6509
     )
 
+if ("dtls" IN_LIST FEATURES)
+    set(ENABLE_DTLS yes)
+else()
+    set(ENABLE_DTLS no)
+endif()
+
+set(CMAKE_C_FLAGS "-DWOLFSSL_ALT_CERT_CHAINS -DWOLFSSL_DES_ECB -DWOLFSSL_CUSTOM_OID -DHAVE_OID_ENCODING -DWOLFSSL_CERT_GEN -DWOLFSSL_ASN_TEMPLATE -DWOLFSSL_KEY_GEN -DHAVE_PKCS7 -DHAVE_AES_KEYWRAP -DWOLFSSL_AES_DIRECT -DHAVE_X963_KDF")
```
Comment (vicroms) [inline ports/wolfssl/portfile.cmake:18]: This needs to call `vcpkg_cmake_get_vars()` to get the combined C/CXX flags and append them to vcpkg's detected configuration.

**Review by vicroms — DISMISSED** [review]

File: ports/wolfssl/portfile.cmake (hunk 1 above)
Comment (laugengebaeck) [reply ports/wolfssl/portfile.cmake:18]: Ah, I didn't want to set the global `CMAKE_C_FLAGS`, but rather the `CMAKE_C_FLAGS` used by `vcpkg_cmake_configure` as an option in line 40 (this is also what the portfile did before). Renaming the variable now, then this should be okay, right?

Comment (laugengebaeck) [reply ports/wolfssl/portfile.cmake]: Fixed this now. Can you review again?

**Comment by jimwang118** [comment]
@laugengebaeck Please resolve code conflicts.

**Comment by laugengebaeck** [comment]
@jimwang118 Conflicts are resolved now, please review.

**Pull request merged**